x-pack/plugins/observability_solution/inventory/public/analytics/register_eem_enabled_context.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AnalyticsServiceSetup } from '@kbn/core/public';
+import { BehaviorSubject } from 'rxjs';
+import { EntityManagerPublicPluginSetup } from '@kbn/entityManager-plugin/public';
+
+export let eemEnabled$: BehaviorSubject<{ eem_enabled: boolean }>;
```

Comment:
>[!TIP]
> I'm on the fence about having a global variable like this. I think the best way would be to encapsulate the `analytics.registerContextProvider` in the TelemetryClient. eg:

```ts
export class TelemetryClient implements ITelemetryClient {
  private eemEnabled$: BehaviorSubject<{ eem_enabled: boolean }>;

  constructor(private analytics: AnalyticsServiceSetup) {
    this.eemEnabled$ = new BehaviorSubject<{ eem_enabled: boolean }>({ eem_enabled: false });
  }

  public initialize = () => {
    this.analytics.registerContextProvider({
      name: 'eem_enabled',
      context$: this.eemEnabled$,
      schema: {
        eem_enabled: {
          type: 'boolean',
          _meta: { description: 'Whether EEM is enabled or not.' },
        },
      },
    });
  };

  public updateEemEnabled = (enabled: boolean) => {
    this.eemEnabled$.next({ eem_enabled: enabled });
  };
```

you could call `telemetry.updateEemEnabled` in the `useEntityManager` hook to update the context for the first time.

Reply:
Thanks for your input.

I based my approach on what we did in this [PR](https://github.com/elastic/kibana/pull/188627) without really questioning it.

The only issue I see with your proposal is that `eem_enabled` won’t be initialized until we use the `useEntityManager` hook, which only happens when the user accesses the Inventory.

The main purpose of populating `eem_enabled` is explained in [this comment](https://github.com/elastic/kibana/issues/195608#issuecomment-2414056790), which states:

> We're basically trying to capture the number of users who are eligible to use the new experience - regardless of whether they visit Inventory or not

Any thoughts on this?

Reply:
I'm adding a reference [here](https://github.com/elastic/kibana/pull/197055#discussion_r1808835671) to a comment I just made regarding the possibility of rethinking this to work as context.

Reply:
You could call `entityManager.entityClient.isManagedEntityDiscoveryEnabled()` and get the needed information. I'm just not sure whether calling an endpoint in the plugin initialization is a good idea.

Reply:
Hey @crespocarlos, thank you so much for clarifying your suggestion.

You can find the changes in this commit https://github.com/elastic/kibana/pull/197055/commits/a4bdbb2e60e50423e68cf6c199aa7a7ede801b81, it’s now much more encapsulated and clear.

Regarding your point about adding that information to analytics within the `EntityManager` plugin, it makes sense, but I don't think we should block the PR to extend the scope to other plugins at the moment.

I still plan to discuss with the team that manages all the telemetry to see what they suggest for cases where we want to get data like this, because from my perspective, adding that info to the context isn’t ideal since it impacts all of Kibana. For now, we've decided to proceed with this approach and add a new attribute to the `context`.

I’m committed to refactoring this once we have clearer guidance from their side, and if they’re okay with adding it to the `context`, I’ll propose setting up telemetry in the `EntityManager` if that makes sense for managing this.

What do you think?

Reply:
Makes sense. Thanks for making the changes. 

Should we ask the analytics folks to review this PR? Just to be on the safe side, to ensure we're not merging anything that could bring problems.

Reply:
I’m marking the conversation as resolved since the `eem_enabled` implementation has been removed.